doc/source/visualisation.rst

```diff
@@ -1,4 +1,145 @@
+ 
+   >>> layout_subgraph = ig.Layout(coords=layout[:2])
+
+It is possible to perform linear transformations to the layout:
+
+- :meth:`Layout.translate`
+- :meth:`Layout.center`
+- :meth:`Layout.scale`
+- :meth:`Layout.fit_into`
+- :meth:`Layout.rotate`
+- :meth:`Layout.mirror`
+
+as well as a generic nonlinear transformation via:
+
+- :meth:`Layout.transform`
+
+The following regular layouts are supported:
+
+- `Graph.layout_star`: star layout
+- `Graph.layout_circle`: circular/spherical layout
+- `Graph.layout_grid`: regular grid layout in 2D
+- `Graph.layout_grid_3d`: regular grid layout in 3D
+- `Graph.layout_random`: random layout (2D and 3D)
+
+The following algorithms produce nice layouts for general graphs:
+
+- `Graph.layout_davidson_harel`: Davidson-Harel layout, based on simulated annealing optimization including edge crossings
+- `Graph.layout_drl`: DrL layout for large graphs (2D and 3D), a scalable force-directed layout
+- `Graph.layout_fruchterman_reingold`: Fruchterman-Reingold layout (2D and 3D), a "spring-electric" layout based on classical physics
+- `Graph.layout_graphopt`: the graphopt algorithm, another force-directed layout
+- `Graph.layout_kamada_kawai`: Kamada-Kawai layout (2D and 3D), a "spring" layout based on classical physics
+- `Graph.layout_lgl`: Large Graph Layout
+- `Graph.layout_mds`: multidimensional scaling layout
+
+The following algorithms are useful for *trees* (and for Sugiyama *directed acyclic graphs* or *DAGs*):
+
+- `Graph.layout_reingold_tilford`: Reingold-Tilford layout
+- `Graph.layout_reingold_tilford_circular`: circular Reingold-Tilford layout
+- `Graph.layout_sugiyama`: Sugiyama layout, a hierarchical layout
+
+For *bipartite graphs*, there is a dedicated function:
+
+- `Graph.layout_bipartite`: bipartite layout
+
+More might be added in the future, based on request.
+
+Graph plotting
+==============
+Once the layout of a graph has been computed, |igraph| can assist with the plotting itself. Plotting happens within a single
+function, `igraph.plot`. This function can be called in two ways:
+
+1. without a filename argument: this generates a temporary file and opens it with the default image viewer:
+
+>>> ig.plot(g)
+
+(see below if you are using this in a `Jupyter`_ notebook).
+
+2. with a filename argument: this stores the graph image in the specified file and does *not* open it automatically. Based on
+   the filename extension, any of the following output formats can be chosen: PNG, PDF, SVG and PostScript:
+
+>>> ig.plot(g, 'myfile.pdf')
+
+.. note:: PNG is a raster image format while PDF, SVG, and Postscript are vector image formats. Choose one of the last three
+   formats if you are planning on refining the image with a vector image editor such as Inkscape or Illustrator.
+
+In both cases, you can add an argument `layout` to the `plot` function to specify a precomputed layout, e.g.:
+
+>>> layout = g.layout("kamada_kawai")
+>>> ig.plot(g, layout=layout)
```

Comment:
It is also possible to just use the name of the layout algorithm in `ig.plot()`, e.g.: `ig.plot(g, layout="kamada_kawai")`. if you don't add a layout, igraph uses the dedicated `layout_auto()` function, which chooses between one of several possible layouts based on the number of vertices and edges.

Reply:
Yep, I added that info now